tests/test_propagator.py

```diff
@@ -44,6 +44,16 @@ def test_4d(self):
         f(data, arr)
         assert(arr[4][3][2][1] == 122)
 
+    def test_space_dims_2d(self):
+        space_dims = (z, x)
+        propagator = Propagator("process", 1, (4, 3), [], space_dims=space_dims)
```

Comment:
This looks like it only tests prescribed `space_dims`. Should we also test the defaulting behaviour?